### PR TITLE
Update Readme

### PR DESCRIPTION
**Pull request by mathieuprog**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -43,17 +59,9 @@ pnpm add solid-styled babel-plugin-solid-styled
...
-Add this to any d.ts file
-
-```ts
-/// <reference types="solid-styled" />
-```
-
 ### `<StyleRegistry>`
 
-`<StyleRegistry>` manages the lifecycle of stylesheets. It is required only to be used once, preferably at the root of your SolidJS application
+`<StyleRegistry>` manages the lifecycle of stylesheets (for instance styles of a component get removed from memory if all instances of that component unmount). It needs to be included only once, preferably at the root of your SolidJS application.
```
Comment (lxsmnsyc) [inline README.md]: It's probably helpful to mention that this only happens on client-side.

File: README.md [hunk 2]
```diff
@@ -257,6 +265,8 @@ function Button() {
 
 You can also use `<style jsx global>` for declaring global styles.
 
+The main motivation for writing an alternative way of declaring styles with `<style jsx>` is to facilitate the migration from `solid-styled-jsx` to `solid-styled`.
```
Comment (lxsmnsyc) [inline README.md]: Probably also mention the familiarity and DX of styled-jsx

File: README.md (hunk 1 above)
Comment (mathieuprog) [reply README.md]: Done

File: README.md (hunk 2 above)
Comment (mathieuprog) [reply README.md]: Done

**Comment by lxsmnsyc** [comment]
Thank you

**Pull request merged**